File: tests/fixtures/hello_service.py
# Long-running service fixture for runtime tests.
import signal
import sys
import time

for i, a in enumerate(sys.argv[1:]):
    print(f"ARG{i}={a}", flush=True)
print("READY", flush=True)


def on_sig(signum, frame):
    print("CLEAN-SHUTDOWN", flush=True)
    sys.exit(0)


signal.signal(signal.SIGINT, on_sig)
n = 0
while True:
    time.sleep(0.3)
    print(f"TICK {n}", flush=True)
    n += 1

File: tests/fixtures/hello_service.js
// Long-running service fixture for runtime tests.
// Prints READY, echoes argv (verifying argument escaping), ticks, and exits
// cleanly on ctrl-c (SIGINT is synthesized from console ctrl events on Windows).
let n = 0;
process.argv.slice(2).forEach((v, i) => console.log(`ARG${i}=${v}`));
console.log('READY');
process.on('SIGINT', () => {
    console.log('CLEAN-SHUTDOWN');
    process.exit(0);
});
setInterval(() => console.log(`TICK ${n++}`), 300);
